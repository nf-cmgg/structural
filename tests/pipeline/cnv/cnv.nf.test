nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_cnv"

    options "-stub"

    test("one caller") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "qdnaseq"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,

                    path("$outputDir")
                        .list()
                        .findAll {
                            !it.toString().endsWith("pipeline_info")
                        }
                        .collect {
                            it.list().collect { it.toString().split("/")[-1] }
                        },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("qdnaseq")}
            )
        }

    }

    test("all callers") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "cnv"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,

                    path("$outputDir")
                        .list()
                        .findAll {
                            !it.toString().endsWith("pipeline_info")
                        }
                        .collect {
                            it.list().collect { it.toString().split("/")[-1] }
                        },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("all")}
            )
        }

    }

}
