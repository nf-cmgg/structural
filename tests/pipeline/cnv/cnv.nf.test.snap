{
    "all": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.cnv.vcf.gz",
                    "PosCon1.cnv.vcf.gz.tbi",
                    "PosCon1.qdnaseq.abberations.bed",
                    "PosCon1.qdnaseq.bed",
                    "PosCon1.qdnaseq.cna",
                    "PosCon1.qdnaseq_segments.txt",
                    "PosCon1.wisecondorx_aberrations.bed",
                    "PosCon1.wisecondorx_bins.bed",
                    "PosCon1.wisecondorx_chr_statistics.txt",
                    "PosCon1.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon2.cnv.vcf.gz",
                    "PosCon2.cnv.vcf.gz.tbi",
                    "PosCon2.qdnaseq.abberations.bed",
                    "PosCon2.qdnaseq.bed",
                    "PosCon2.qdnaseq.cna",
                    "PosCon2.qdnaseq_segments.txt",
                    "PosCon2.wisecondorx_aberrations.bed",
                    "PosCon2.wisecondorx_bins.bed",
                    "PosCon2.wisecondorx_chr_statistics.txt",
                    "PosCon2.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon3.cnv.vcf.gz",
                    "PosCon3.cnv.vcf.gz.tbi",
                    "PosCon3.qdnaseq.abberations.bed",
                    "PosCon3.qdnaseq.bed",
                    "PosCon3.qdnaseq.cna",
                    "PosCon3.qdnaseq_segments.txt",
                    "PosCon3.wisecondorx_aberrations.bed",
                    "PosCon3.wisecondorx_bins.bed",
                    "PosCon3.wisecondorx_chr_statistics.txt",
                    "PosCon3.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon4.cnv.vcf.gz",
                    "PosCon4.cnv.vcf.gz.tbi",
                    "PosCon4.qdnaseq.abberations.bed",
                    "PosCon4.qdnaseq.bed",
                    "PosCon4.qdnaseq.cna",
                    "PosCon4.qdnaseq_segments.txt",
                    "PosCon4.wisecondorx_aberrations.bed",
                    "PosCon4.wisecondorx_bins.bed",
                    "PosCon4.wisecondorx_chr_statistics.txt",
                    "PosCon4.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "multiqc_data",
                    "multiqc_plots",
                    "multiqc_report.html"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "PosCon4",
                "multiqc",
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-13T11:31:18.794715638"
    },
    "qdnaseq": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.cnv.vcf.gz",
                    "PosCon1.cnv.vcf.gz.tbi",
                    "PosCon1.qdnaseq.abberations.bed",
                    "PosCon1.qdnaseq.bed",
                    "PosCon1.qdnaseq.cna",
                    "PosCon1.qdnaseq_segments.txt",
                    "statistics.out"
                ],
                [
                    "PosCon2.cnv.vcf.gz",
                    "PosCon2.cnv.vcf.gz.tbi",
                    "PosCon2.qdnaseq.abberations.bed",
                    "PosCon2.qdnaseq.bed",
                    "PosCon2.qdnaseq.cna",
                    "PosCon2.qdnaseq_segments.txt",
                    "statistics.out"
                ],
                [
                    "PosCon3.cnv.vcf.gz",
                    "PosCon3.cnv.vcf.gz.tbi",
                    "PosCon3.qdnaseq.abberations.bed",
                    "PosCon3.qdnaseq.bed",
                    "PosCon3.qdnaseq.cna",
                    "PosCon3.qdnaseq_segments.txt",
                    "statistics.out"
                ],
                [
                    "PosCon4.cnv.vcf.gz",
                    "PosCon4.cnv.vcf.gz.tbi",
                    "PosCon4.qdnaseq.abberations.bed",
                    "PosCon4.qdnaseq.bed",
                    "PosCon4.qdnaseq.cna",
                    "PosCon4.qdnaseq_segments.txt",
                    "statistics.out"
                ],
                [
                    "multiqc_data",
                    "multiqc_plots",
                    "multiqc_report.html"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "PosCon4",
                "multiqc",
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-13T11:28:36.085432115"
    }
}